obspy/io/gcf/libgcf.py

```
@@ -0,0 +1,159 @@
+    s = ''
+    while data:
+        imed = data % 36
+        if imed > 9:
+            imed += 7
+        s = chr(imed + 48) + s
+        data = np.uint32(data / 36)
+    return s
+
+
+def decode_date_time(data):
+    """
+    Decode date and time field.
+
+    The date code is a 32 bit value specifying the start time of the block.
+    Bits 0-16 contain the number of seconds since midnight,
+    and bits 17-31 the number of days since 17th November 1989.
+    """
+    days = data >> 17
+    secs = data & 0x1FFFF
+    starttime = UTCDateTime('1989-11-17') + days * 86400 + secs
+    return starttime
+
+
+def read_data_block(f, headonly=False, channel_prefix="HH", **kwargs):
+    """Read one data block from GCF file.
+
+    more details can be found here:
+    http://geophysics.eas.gatech.edu/GTEQ/Scream4.4/GCF_Specification.htm
+    f - file object to read from
+    if skipData is True, Only header is returned.
+    if not a data block (SPS=0) - returns None.
+    """
+    # get ID
+    sysid = f.read(4)
+    if not sysid:
+        raise EOFError  # got to EOF
+    sysid = np.frombuffer(sysid, count=1, dtype='>u4')
+    if sysid >> 31 & 0b1 > 0:
+        sysid = (sysid << 6) >> 6
+    sysid = decode36(sysid)
+    # get Stream ID
+    stid = np.frombuffer(f.read(4), count=1, dtype='>u4')
+    stid = decode36(stid)
+    # get Date & Time
+    data = np.frombuffer(f.read(4), count=1, dtype='>u4')
+    starttime = decode_date_time(data)
+    # get data format
+    # get reserved, SPS, data type compression, number of 32bit records (N)
+    reserved, sps, compress, N = np.frombuffer(f.read(4), count=4, dtype='>u1')
+    compression = compress & 0b00001111  # get compression code
+    t_offset = compress >> 4  # get time offset
+    if t_offset > 0:
+        starttime = starttime + t_offset / time_offsets_d[sps]
+    if sps in sps_d:
+        sps = sps_d[sps]  # get special SPS value if needed
```

Are there any non-special values that are invalid that should raise an exception here?